dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -2611,225 +2612,65 @@ public boolean isContentEqual(Contentlet contentlet1,Contentlet contentlet2, Use
-                contentRelationships = findContentRelationshipsFromIndex(workingCon);
-            } else {
-                contentRelationships = findContentRelationshipsFromIndex(contentlet);
-            }
-        } else {
-            contentRelationships = findContentRelationshipsFromIndex(contentlet);
-        }
-
-        if(cats == null)
-            cats = new ArrayList<Category>();
-        if(contentRelationships == null)
-            contentRelationships = new HashMap<Relationship, List<Contentlet>>();
-        if(workingCon == null)
-            workingCon = contentlet;
-        return checkin(contentlet, contentRelationships, cats, permissions, user, respectFrontendRoles);
+    public Contentlet checkin(Contentlet contentlet, List<Permission> permissions, User user,
+                              boolean respectFrontendRoles)
+        throws IllegalArgumentException,DotDataException, DotSecurityException,DotContentletStateException {
+        return checkin(contentlet, (ContentletRelationships) null, null, permissions, user, respectFrontendRoles);
     }
 
     @CloseDBIfOpened
     @Override
-    public Contentlet checkin(Contentlet contentlet,Map<Relationship, List<Contentlet>> contentRelationships,List<Category> cats, User user, boolean respectFrontendRoles)throws IllegalArgumentException, DotDataException,DotSecurityException, DotContentletStateException,DotContentletValidationException {
-
-        List<Permission> permissions = null;
-        Contentlet workingCon = null;
-
-        //If contentlet is not new
-        if(InodeUtils.isSet(contentlet.getIdentifier())) {
-            workingCon = findWorkingContentlet(contentlet);
-            if(workingCon != null){
-                if(cats==null) {
-                    cats = categoryAPI.getParents(workingCon, APILocator.getUserAPI().getSystemUser(), true);
-                }
-                if(contentRelationships==null) {
-                    contentRelationships = findContentRelationships(workingCon);
-                }
-                permissions = permissionAPI.getPermissions(workingCon);
-            }
-        }
-
-        if(permissions == null)
-            permissions = new ArrayList<Permission>();
-
-        if(cats == null)
-            cats = new ArrayList<Category>();
-        if(contentRelationships == null)
-            contentRelationships = new HashMap<Relationship, List<Contentlet>>();
-        if(workingCon == null)
-            workingCon = contentlet;
-        return checkin(contentlet, contentRelationships, cats, permissions, user, respectFrontendRoles);
+    public Contentlet checkin(Contentlet contentlet,Map<Relationship, List<Contentlet>> contentRelationships,
+                              List<Category> cats, User user, boolean respectFrontendRoles)
+        throws IllegalArgumentException, DotDataException,DotSecurityException, DotContentletStateException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.dotmarketing.portlets.contentlet.business.DotContentletStateException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)